src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs

```diff
@@ -189,26 +194,109 @@ public async Task StoreDirectoryAsync(string directoryName, string directoryPath
         {
             try
             {
-                async Task DirectorySearch(string path, string subdir = "")
+                TransferUtilityUploadDirectoryRequest request = new()
+                {
+                    BucketName = _options.BlobStorageName,
+                    Directory = directoryPath,
+                    KeyPrefix = directoryName,
+                    SearchOption = SearchOption.AllDirectories,
+                    UploadFilesConcurrently = true,
+                };
+
+                request.UploadDirectoryFileRequestEvent += (sender, e) =>
                 {
-                    foreach (var file in Directory.GetFiles(path))
+                    e.UploadRequest.CannedACL = (storageObject?.IsPublic ?? false) ? S3CannedACL.PublicRead : S3CannedACL.Private;
+                    e.UploadRequest.Headers.ContentType = storageObject?.ContentType ?? e.UploadRequest.Headers.ContentType;
+                    e.UploadRequest.Headers.CacheControl = storageObject?.CacheControl ?? e.UploadRequest.Headers.CacheControl;
+                    e.UploadRequest.Headers.ContentDisposition = storageObject?.ContentDisposition ?? e.UploadRequest.Headers.ContentDisposition;
+                    e.UploadRequest.Headers.ContentEncoding = storageObject?.ContentEncoding ?? e.UploadRequest.Headers.ContentEncoding;
+                };
+
+                TransferUtilityConfig config = new()
+                {
+                    /** Note: This is currently set to the default value of 10. I did some benchmarking on 23 dec 2020 
```

Comment:
We're using XDC docs and C/C++ comment. Use either of those, not javadoc. Remove name.